Loop TodayExtension/TodayExtensionContext.swift

```diff
@@ -0,0 +1,154 @@
+            
+            var sensor: SensorDisplayableContext? = nil
+            if data["gcsv"] != nil {
+                sensor = SensorDisplayableContext(
+                    isStateValid: data["gcsv"] as! Bool,
+                    stateDescription: data["gcsd"] as! String,
+                    trendType: data["gcst"] as? GlucoseTrend,
+                    isLocal: data["gcsl"] as! Bool)
+            }
+            return GlucoseContext(
+                latest: HKQuantitySample(
+                    type: HKQuantityType.quantityType(forIdentifier: HKQuantityTypeIdentifier.bloodGlucose)!,
+                    quantity: HKQuantity(unit: HKUnit.milligramsPerDeciliterUnit(),
+                                         doubleValue: data["gcgv"] as! Double),
+                    start: data["gcgd"] as! Date,
+                    end: data["gcgd"] as! Date),
+                sensor: sensor)
+        }
+        set(lgv) {
+            data["gcgv"] = lgv?.latest.quantity.doubleValue(for: HKUnit.milligramsPerDeciliterUnit())
+            data["gcgd"] = lgv?.latest.startDate
+            data["gcsv"] = lgv?.sensor?.isStateValid
+            data["gcsd"] = lgv?.sensor?.stateDescription
+            data["gcst"] = lgv?.sensor?.trendType
+            data["gcsl"] = lgv?.sensor?.isLocal
+        }
+    }
+    
+    var batteryPercentage: Double? {
+        get {
+            if data["bp"] == nil { return nil }
+            return data["bp"] as? Double
+        }
+        set(bp) {
+            data["bp"] = bp
+        }
+    }
+    
+    var reservoir: ReservoirValueContext? {
+        get {
+            if data["rsd"] == nil { return nil }
+            return ReservoirValueContext(
+                startDate: data["rsd"] as! Date,
+                unitVolume: data["ruv"] as! Double,
+                capacity: data["rc"] as! Int)
+        }
+        set(rvc) {
+            data["rsd"] = rvc?.startDate
+            data["ruv"] = rvc?.unitVolume
+            data["rc"] = rvc?.capacity
+        }
+    }
+    
+    var loop: LoopContext? {
+        get {
+            if data["lde"] == nil { return nil }
+            return LoopContext(
+                dosingEnabled: data["lde"] as! Bool,
+                lastCompleted: data["llc"] as! Date?)
+        }
+        set(l) {
+            data["lde"] = l?.dosingEnabled
+            data["llc"] = l?.lastCompleted
+        }
+    }
+    
+    var netBasal: NetBasalContext? {
+        get {
+            if data["nbr"] == nil { return nil }
+            return NetBasalContext(
+                rate: data["nbr"] as! Double,
+                percentage: data["nbp"] as! Double,
+                startDate: data["nbsd"] as! Date)
+        }
+        set(b) {
+            data["nbr"] = b?.rate
+            data["nbp"] = b?.percentage
+            data["nbsd"] = b?.startDate
+        }
+    }
+    
+    var eventualGlucose: String? {
+        get {
+            return data["eg"] as? String
```

Comment:
Don't use cryptic dictionary keys here. Match the property names exactly. I chose to use these on the watch dictionaries because they're being shuttled over a low-bandwidth Bluetooth connection.